pkg/snclient/check_logfile.go

```diff
@@ -0,0 +1,256 @@
+		}
+		if inode == parsedFile.inode {
+			parsedFile.offset = 0
+		}
+	}
+
+	// Jump to last read bytes
+	_, err = file.Seek(int64(snc.alreadyParsedLogfiles[fileName].offset), 0)
+	if err != nil {
+		return 0, fmt.Errorf("while skipping already read file an error occurred: %s", err.Error())
+	}
+
+	scanner := bufio.NewScanner(file)
+
+	scanner.Split(c.getCustomSplitFunction())
+	okReset := len(check.okThreshold) > 0
+	lineStorage := make([]map[string]string, 0)
+	var lineIndex int
+
+	// filter each line
+	for lineIndex = 0; scanner.Scan(); lineIndex++ {
+		line := scanner.Text()
+		entry := map[string]string{
+			"filename": fileName,
+			"line":     line,
+		}
+		// We have n Labels that all somehow need to be accessed
+		// We have n Labels that all need to check on each line
+		for label, pattern := range labels {
+			entry[label] = pattern.FindString(line)
+		}
+
+		// get all Thresholds with prefix coulumn
+		// if len(thres) > 0
+		allThresh := append(check.warnThreshold, check.critThreshold...)
+		var columnNumbers []int
+		// Extract all needed threshold number
+		numReg := regexp.MustCompile(`\d+`)
+
+		for _, thresh := range allThresh {
+			if !strings.HasPrefix(thresh.keyword, "column") {
+				continue
+			}
+			match := numReg.FindString(thresh.keyword)
+			if match == "" {
+				continue
+			}
+			var index int
+			index, err = strconv.Atoi(match)
+			if err != nil {
+				return 0, fmt.Errorf("could not extract coulumn number from argument err: %s", err.Error())
+			}
+			columnNumbers = append(columnNumbers, index)
+		}
+
+		if len(columnNumbers) > 0 {
+			cols := strings.Split(line, c.ColumnDelimter)
+			var maxColoumns int
+			if len(columnNumbers) == 0 {
+				maxColoumns = 0
+			} else {
+				maxColoumns = slices.Max(columnNumbers)
+			}
+
+			if len(cols) <= maxColoumns {
+				return 0, fmt.Errorf("not enough columns in log for separator and index")
+			}
+
+			// in range of number of coulumns
+			// Fill entryp map
+			for _, columnIndex := range columnNumbers {
+				entry[fmt.Sprintf("column%d", columnIndex)] = cols[columnIndex]
+			}
+		}
+
+		lineStorage = append(lineStorage, entry)
+		// Do not check for OK with empty conditionlist, it would match all
+		if okReset && check.MatchMapCondition(check.okThreshold, entry, true) {
+			// Add and empty entry with the current line count to the listdata to keep track of linecount
+			entry := map[string]string{
+				"_count": fmt.Sprintf("%d", len(lineStorage)),
+			}
+			check.listData = append(check.listData, entry)
+			lineStorage = make([]map[string]string, 0)
+		}
+	}
+	check.listData = append(check.listData, lineStorage...)
+	// Save File Size to check if lines were added
+	info, err := file.Stat()
+	if err != nil {
+		return 0, fmt.Errorf("could not get file information %s", err.Error())
+	}
+	pf := ParsedFile{path: fileName, offset: int(info.Size())}
+	if runtime.GOOS == "linux" {
+		pf.inode = getInode(fileName)
+	}
+	snc.alreadyParsedLogfiles[fileName] = pf
```

Comment:
this might be an problem if multiple concurrent go routines try to access the map at the same time. Better use a `sync.Map` here or add a `sync.RWMutex`